google-datacatalog-sisense-connector/src/google/datacatalog_connectors/sisense/prepare/assembled_entry_factory.py

```diff
@@ -36,22 +37,54 @@ def __init__(self, project_id: str, location_id: str, entry_group_id: str,
                 project_id, location_id, entry_group_id, user_specified_system,
                 server_address)
 
-    def make_assembled_entries_for_folder(
-            self, folder_metadata: Dict[str, Any]) -> List[AssembledEntryData]:
+        self.__datacatalog_tag_factory = \
+            datacatalog_tag_factory.DataCatalogTagFactory(server_address)
+
+    def make_assembled_entries_list(
+            self, asset_metadata: Dict[str, Any],
+            tag_templates: Dict[str, TagTemplate]) -> List[AssembledEntryData]:
+
+        assembled_entries = []
+        if constants.SISENSE_ASSET_TYPE_FOLDER == asset_metadata.get('type'):
+            assembled_entries.extend(
+                self.__make_assembled_entries_for_folder(
+                    asset_metadata, tag_templates))
+
+        return assembled_entries
+
+    def __make_assembled_entries_for_folder(
+            self, folder_metadata: Dict[str, Any],
+            tag_templates: Dict[str, TagTemplate]) -> List[AssembledEntryData]:
+
+        folder_tag_template = tag_templates.get(
+            constants.TAG_TEMPLATE_ID_FOLDER)
 
         assembled_entries = [
-            self.__make_assembled_entry_for_folder(folder_metadata)
+            self.__make_assembled_entry_for_folder(folder_metadata,
+                                                   folder_tag_template)
         ]
 
+        for child_folder in folder_metadata.get('folders') or []:
+            assembled_entries.append(
+                self.__make_assembled_entry_for_folder(child_folder,
```

Comment:
Can you have more of one level of child folders? if you can and it's not being handled, please add this info to the docs. Otherwise you can ignore my comment.

Reply:
There was an error in the code, thanks for finding it! 👏🏽 
I've replaced with:
```python
    assembled_entries.extend(
        self.__make_assembled_entries_for_folder(
            child_folder, tag_templates))
```
Now, `__make_assembled_entries_for_folder()` is called recursively and all child folders are processed accordingly.